connectors/aws-cert/src/main/java/com/vmware/connectors/aws/cert/AwsCertController.java

```diff
@@ -0,0 +1,331 @@
+import com.vmware.connectors.common.payloads.response.Card;
+import com.vmware.connectors.common.payloads.response.CardAction;
+import com.vmware.connectors.common.payloads.response.CardActionKey;
+import com.vmware.connectors.common.payloads.response.CardBody;
+import com.vmware.connectors.common.payloads.response.CardBodyField;
+import com.vmware.connectors.common.payloads.response.CardBodyFieldType;
+import com.vmware.connectors.common.payloads.response.Cards;
+import com.vmware.connectors.common.utils.Async;
+import com.vmware.connectors.common.utils.CardTextAccessor;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.FormElement;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.AsyncRestOperations;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+import rx.Observable;
+import rx.Single;
+
+import javax.validation.Valid;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+public class AwsCertController {
+
+    private static final String APPROVAL_URL_PARAM = "hero_aws_cert_approval_url";
+
+    private static final Logger logger = LoggerFactory.getLogger(AwsCertController.class);
+
+    private static final String ROUTING_PREFIX = "x-routing-prefix";
+
+    private static final String APPROVE_PATH = "/api/v1/approve";
+
+    private final String certificateApprovalHost;
+    private final String certificateApprovalPath;
+    private final AsyncRestOperations rest;
+    private final CardTextAccessor cardTextAccessor;
+
+    @Autowired
+    public AwsCertController(
+            @Value("${aws.certificate.connector.approval.host}") String certificateApprovalHost,
```

Comment:
This host and path are used to prevent abuse (we don't want to allow a client to tell the server to GET/POST to anywhere they want).